spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,145 @@
+    """
+
+    @monitors.name('Should extract the minimum amount of items')
+    def test_minimum_number_of_items(self):
+        minimum_threshold = self.get_settings(SPIDERMON_MIN_ITEMS)
+        if not minimum_threshold:
+            raise NotConfigured('You should specify a minimum number of items '
+                                'to check against.')
+
+        item_extracted = getattr(self.stats, 'item_scraped_count', 0)
+        msg = 'Extracted {} items, the expected minimum is {}'.format(
+            item_extracted, minimum_threshold)
+        self.assertTrue(
+            item_extracted >= minimum_threshold, msg=msg
+        )
+
+
+@monitors.name('Log Monitor')
+class LogMonitor(BaseScrapyMonitor):
+    """ Check for errors in the spider log you can configure the expected
+    number of ERROR log messages using ``SPIDERMON_MAX_ERROR``,
+    the default is ``0``."""
+
+    @monitors.name('Should not have any errors')
+    def test_should_not_have_errors(self):
+        errors_threshold = self.get_settings(SPIDERMON_MAX_ERROR, 0)
+        no_of_errors = self.stats.get('log_count/ERROR', 0)
+        msg = 'Found {} errors in log, maximum expected is '\
+              '{}'.format(no_of_errors, errors_threshold)
+        self.assertTrue(no_of_errors <= errors_threshold, msg=msg)
+
+
+@monitors.name('Finish Reason Monitor')
+class FinishReasonMonitor(BaseScrapyMonitor):
+    """ Check if a job has a expected finish reason.
+    You can configure the expected reason with the
+    ``SPIDERMON_EXPECTED_FINISH_REASONS``, it should be an ``iterable`` of
+    valid finish reasons.
+
+    The default value of this settings is: ``['finished', ]``"""
+
+    @monitors.name('Should have the expected finished reason(s)')
+    def test_should_not_have_errors(self):
+        expected_reasons = self.get_settings(
+            SPIDERMON_EXPECTED_FINISH_REASONS, ('finished', ))
+        finished_reason = self.stats.get('finish_reason')
+        msg = 'Finished with "{}" the expected reasons are {}'.format(
+                finished_reason, expected_reasons)
+        self.assertTrue(finished_reason in expected_reasons, msg=msg)
+
+
+@monitors.name('Check for unwanted http status codes')
+class UnwantedHttpStatus(BaseScrapyMonitor):
+    """ Check for maximum number of unwanted http codes.
+
+    You can configure a list of unwanted http codes with
+    ``SPIDERMON_UNWANTED_HTTP_CODES`` the default value is:
+    ``[400, 407, 429, 500, 502, 503, 504, 523, 540, 541]``
+
+    You can also configured the **maximum** number of times that those
+    response codes with ``SPIDERMON_MAX_UNWANTED_HTTP_CODES`` the default
+    value is ``1``.
+    """
+    default_codes = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+
+    @monitors.name('Should not hit the limit of unwanted http status')
+    def test_check_unwanted_http_codes(self):
+        max_errors = self.get_settings(SPIDERMON_MAX_UNWANTED_HTTP_CODES, 1)
```

Comment:
What do you think of a threshold for each error code?
Probably, I accept up to 10 404s, but up to 50 503s.

Reply:
Also, not sure if `max_errors` or `max_status_count`.
An HTTP status code is not necessarily an error.

Reply:
We need this monitor for sure, if this is the question here.
Establishing arbitrary values about what I think should be the threshold is a bad idea, it should be flexible enough so you can configure the way you want in your project.

Reply:
You're talking about having something like:

``python
SPIDERMON_UNWANTED_HTTP_CODES = {
    '500: 50,
    '400': 10,
}``

I like the flexibility of the approach but we actually need it? You can't have a threshold to rule then all?
Would like to hear more comments on this one as well.

Reply:
It would be good but we can live without it for a while.

Reply:
It would be good but we can live without it for a while.

Reply:
@rennerocha and @raphapassini  This was my idea, sorry if I wasn't clear enough :)
For me, the default threshold for all error codes are fine, just wrote a thought that could be helpful to make it more flexible.
Agree with @victor-torres and @rennerocha  that we can live with a single value for all codes for now, and in the future we could change it.

Reply:
I like the suggestion it was implemented :)